components/esp-tls/esp_tls_mbedtls.c

```diff
@@ -787,6 +796,14 @@ int esp_mbedtls_server_session_create(esp_tls_cfg_server_t *cfg, int sockfd, esp
         tls->conn_state = ESP_TLS_FAIL;
         return -1;
     }
+
+#if defined(CONFIG_ESP_TLS_SERVER_SNI_HOOK)
+    if (cfg->sni_callback != NULL) {
+        ESP_LOGI(TAG, "Initializing server side SNI callback");
+        mbedtls_ssl_conf_sni(&tls->conf, cfg->sni_callback, cfg->sni_callback_p_info);
+    }
+#endif
```

Comment:
This part should come under `set_server_config` API  since this part also does server configuration.
I think we can keep this `conf_sni` part in the `else` condition added above.

Reply:
I'm not sure this is the case. The set_server_config is called when a new server side socket is created, while server_session_create is called each time a client connection is accepted. It is at that  point that SNI has to be initialized, per incoming socket.

Adding it to set_server_config did not actually activate SNI as far as I remember - that's what I tried first.

Reply:
@axos88 Sorry for the delayed reply, I wanted to test this first at my end before commenting.
I will do that now and give my comments in some time.
Do you have a MWE which I could use to test out this feature ?
Thanks!

Reply:
Hi @axos88 I was able to test the SNI feature locally.
I can confirm that with current approach if `servercert` and `server key` are not supplied and we configure the sni_callback using `mbedtls_ssl_conf_sni` in the `ser_server_config` function. Then the sni callback is set at once and is used for every connection formed by that specific context. There is no need to set the sni callback each time in the `esp_mbedtls_server_session_create`. I will update my comments based on that.